### PR TITLE
upgrade mypy

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-4280

## Description
- Upgrading mypy
- Do not lint with `mypy` on python2
- Show traceback of `mypy` to locate `mypy` bugs in the build
- Default to run type checking with site packaging/

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/lint/linter.py [hunk 1]
```diff
@@ -305,7 +304,7 @@ def _gather_facts(self, modules: dict) -> bool:
         self._split_lint_files()
 
         self._linter_to_commands()
-
+
```
Comment (tkatzir) [inline demisto_sdk/commands/lint/linter.py:307]: Extra tab?

File: demisto_sdk/commands/lint/commands_builder.py [hunk 2]
```diff
@@ -170,8 +170,10 @@ def build_mypy_command(files: List[Path], version: str, content_repo: Path = Non
     command += " --allow-redefinition"
     # Get the full path to the file.
     command += " --show-absolute-path"
-    # Ignore site packages because running on host
-    command += " --no-site-packages"
```
[the remark this answers is not in the document]
Comment (ilaner) [reply demisto_sdk/commands/lint/commands_builder.py:174]: The reason we added this parameter is for our nightly. Without this parameter, the type checking is more strict and will check the types of the packages used.
As the `mypy` upgrade makes a lot of packages to fail, for the migration we will require those issues to be fixed in the personal build only, and the developers should fix those issues when merging to content.

File: demisto_sdk/commands/lint/linter.py [hunk 3]
```diff
@@ -368,7 +367,8 @@ def _run_lint_in_host(self, no_bandit: bool, no_mypy: bool,
                 elif lint_check == "bandit" and not no_bandit:
                     exit_code, output = self._run_bandit(lint_files=self._facts["lint_files"])
 
-                elif lint_check == "mypy" and not no_mypy:
+                elif lint_check == "mypy" and not no_mypy and self._facts['python_version'] == '3':
```
[the remark this answers is not in the document]
Comment (ilaner) [reply demisto_sdk/commands/lint/linter.py]: it is a string. Maybe we can do 
```int(self._facts['python_version'][0]) >= 3```

File: demisto_sdk/commands/lint/linter.py [hunk 4]
```diff
@@ -247,7 +247,6 @@ def _gather_facts(self, modules: dict) -> bool:
                     logger.info(f"{self._pack_name} - Facts - {image[0]} - Python {py_num}")
                     if not self._facts["python_version"]:
                         self._facts["python_version"] = py_num
-
```
Comment (yaakovi) [inline demisto_sdk/commands/lint/linter.py:250]: restore

File: demisto_sdk/commands/lint/linter.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply demisto_sdk/commands/lint/linter.py]: use the `Version` object and check `version.major`

File: demisto_sdk/commands/lint/commands_builder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply demisto_sdk/commands/lint/commands_builder.py:174]: Need to document it

**Comment by ilaner** [comment]
@yaakovi 
Example of summary report

**Review by tkatzir — APPROVED** [review]

**Pull request merged**